test/metabase/search/postgres/index_test.clj

```diff
@@ -46,6 +47,51 @@
       (search.ingestion/populate-index!)
       (is (= rows-before (count-rows))))))
 
+(deftest incremental-update-test
+  (with-index
+   (testing "The index is updated when models change"
+     ;; The second entry is "Revenue Project(ions)"
+     (is (= 2 (count (search.index/search "Projected Revenue"))))
+     (is (= 0 (count (search.index/search "Protected Avenue"))))
+
+     (t2/update! :model/Card {:name "Projected Revenue"} {:name "Protected Avenue"})
+     ;; TODO wire up an actual hook
+     (search.ingestion/update-index! (t2/select-one :model/Card :name "Protected Avenue"))
+
+     (is (= 1 (count (search.index/search "Projected Revenue"))))
+     (is (= 1 (count (search.index/search "Protected Avenue"))))
+
+     ;; TODO wire up the actual hook, and actually delete it
+     (search.ingestion/delete-model! (t2/select-one :model/Card :name "Protected Avenue"))
+
+     (is (= 1 (count (search.index/search "Projected Revenue"))))
+     (is (= 0 (count (search.index/search "Protected Avenue")))))))
+
+(deftest related-update-test
+  (with-index
+   (testing "The index is updated when model dependencies change"
+     (let [index-table    @#'search.index/active-table
+           table-idx      (t2/select-one [index-table :id :model_id] :model "table")
```

Comment:
This is pretty dodgy. It'll be better to add a new database and table to the `with-index` fixtures. That way we don't need to worry about clean-up either.